WooCommerce/WooCommerceUITests/Screens/MyStore/MyStoreScreen.swift

```diff
@@ -1,39 +1,39 @@
-import Foundation
+import ScreenObject
 import XCTest
 
-final class MyStoreScreen: BaseScreen {
-
-    struct ElementStringIDs {
-        static let topBannerCloseButton = "top-banner-view-dismiss-button"
-        static let settingsButton = "dashboard-settings-button"
-    }
+final class MyStoreScreen: ScreenObject {
 
     let tabBar = TabNavComponent()
     let periodStatsTable = PeriodStatsTable()
-    private let settingsButton: XCUIElement
-    private let topBannerCloseButton: XCUIElement
 
-    static var isVisible: Bool {
-        let settingsButton = XCUIApplication().buttons[ElementStringIDs.settingsButton]
-        return settingsButton.exists && settingsButton.isHittable
+    private let settingsButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons["dashboard-settings-button"]
+    }
+
+    private let topBannerCloseButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons["top-banner-view-dismiss-button"]
     }
 
-    init() {
-        settingsButton = XCUIApplication().buttons[ElementStringIDs.settingsButton]
-        topBannerCloseButton = XCUIApplication().buttons[ElementStringIDs.topBannerCloseButton]
+    private var settingsButton: XCUIElement { settingsButtonGetter(app) }
+    private var topBannerCloseButton: XCUIElement { topBannerCloseButtonGetter(app) }
 
-        super.init(element: settingsButton)
+    static var isVisible: Bool {
+        guard let screen = try? MyStoreScreen() else { return false }
+        return screen.settingsButton.isHittable
+    }
```

Comment:
This is the same functionality as the implementation above. Initializing a `ScreenObject` already ensures it `.exists`, throwing an error otherwise.

With `try?` we can capture a thrown error and "pretend" that instead of receiving a value from the `init` method (`MyStoreScreen()`) we got `.none`. The `guard let screen...` will fail if the value is `.none` and run the `else` branch.